packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -26,6 +27,31 @@ export class AssociationManager {
     };
   }
 
+  public static addEventNotifications(portfolio: IPortfolio, product: IProduct, topics: sns.ITopic[], options: TagUpdateConstraintOptions): void {
+    if (!topics.length) {
+      throw new Error(`No topics provided for launch notifications for association ${this.prettyPrintAssociation(portfolio, product)}`);
+    }
+    InputValidator.validateLength(this.prettyPrintAssociation(portfolio, product), 'description', 0, 2000, options.description);
+    const association = this.associateProductWithPortfolio(portfolio, product);
+
+    for (const topic of topics) {
+      const constructId = `LaunchNotificationConstraint${hashValues(topic.node.addr, association.associationKey)}`;
```

Comment:
Should `topic.stack.node.addr` be added here?

Reply:
it can be added.